src/WebJobs.Extensions.OpenAI/Embeddings/EmbeddingsHelper.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.AI.OpenAI;
+
+namespace Microsoft.Azure.WebJobs.Extensions.OpenAI.Embeddings;
+static class EmbeddingsHelper
+{
+    static readonly char[] sentenceEndingsDefault = new[] { '.', '!', '?' };
+    static readonly char[] wordBreaksDefault = new[] { ',', ';', ':', ' ', '(', ')', '[', ']', '{', '}', '\t', '\n' };
+
+    public static EmbeddingsOptions BuildRequest(int maxOverlap, int maxChunkLength, string model, InputType inputType, string input)
+    {
+        using TextReader reader = GetTextReader(inputType, input);
+        if (maxOverlap >= maxChunkLength)
+        {
+            throw new ArgumentOutOfRangeException($"MaxOverlap ({maxOverlap}) must be less than MaxChunkLength ({maxChunkLength}).");
+        }
+
+        List<string> chunks = GetTextChunks(reader, 0, maxChunkLength, maxOverlap).ToList();
+        return new EmbeddingsOptions(model, chunks);
+    }
+
+    static TextReader GetTextReader(InputType inputType, string input)
+    {
+        if (inputType == InputType.RawText)
+        {
+            return new StringReader(input);
+        }
+        else if (inputType == InputType.FilePath)
+        {
+            return new StreamReader(input);
+        }
+        else if (inputType == InputType.URL)
+        {
+            HttpClient client = new();
```

Comment:
We mustn't create a new `HttpClient` for each request or we'll run into port exhaustion issues. Instead, we should accept an `IHttpClientFactory` via dependency injection and use that to construct an `HttpClient`. More info here: https://learn.microsoft.com/azure/azure-functions/errors-diagnostics/sdk-rules/azf0002.

Reply:
Also, for better traceability, we should configure the client to add a default `User-Agent` request header so that users can tell when HTTP requests are generated by our extension. Something like `<PackageName>/<PackageVersion>` (e.g. `Microsoft.Azure.WebJobs.Extensions.OpenAI/0.15.0`) would be ideal. Something like this can be used to generate and cache this value:
```csharp
static readonly string UserAgent = $"{typeof(OpenAIExtension).Namespace}/{FileVersionInfo.GetVersionInfo(typeof(OpenAIExtension).Assembly.Location).FileVersion}";
```

Reply:
updated according to both suggestions, please review again